src/Lib.P2P/Cid.cs

```diff
@@ -0,0 +1,610 @@
+        public static Cid Read(byte[] buffer)
+        {
+            var cid = new Cid();
+            if (buffer.Length == 34)
+            {
+                cid.Version = 0;
+                cid.Hash = new MultiHash(buffer);
+                return cid;
+            }
+
+            using (var ms = new MemoryStream(buffer, false))
+            {
+                cid.Version = ms.ReadVarint32();
+                cid.ContentType = ms.ReadMultiCodec().Name;
+                cid.Hash = new MultiHash(ms);
+                return cid;
+            }
+        }
+
+        /// <summary>
+        ///   Returns the binary representation of the CID as a byte array.
+        /// </summary>
+        /// <returns>
+        ///   A new buffer containing the CID.
+        /// </returns>
+        /// <remarks>
+        ///   The buffer does NOT start with a varint length prefix.
+        /// </remarks>
+        public byte[] ToArray()
+        {
+            if (Version == 0) return Hash.ToArray();
+
+            using (var ms = new MemoryStream())
+            {
+                ms.WriteVarint(Version);
+                ms.WriteMultiCodec(ContentType);
+                Hash.Write(ms);
+                return ms.ToArray();
+            }
+        }
+
+        /// <summary>
+        ///   Implicit casting of a <see cref="MultiHash"/> to a <see cref="Cid"/>.
+        /// </summary>
+        /// <param name="hash">
+        ///   A <see cref="MultiHash"/>.
+        /// </param>
+        /// <returns>
+        ///   A new <see cref="Cid"/> based on the <paramref name="hash"/>.  A <see cref="Version"/> 0
+        ///   CID is returned if the <paramref name="hash"/> is "sha2-356"; otherwise <see cref="Version"/> 1.
+        /// </returns>
+        public static implicit operator Cid(MultiHash hash)
+        {
+            if (hash.Algorithm.Name == "sha2-256")
+                return new Cid
+                {
+                    Hash = hash,
+                    Version = 0,
+                    Encoding = "base58btc",
+                    ContentType = "dag-pb"
+                };
+
+            return new Cid
+            {
+                Version = 1,
+                Hash = hash
+            };
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode() { return Encode().GetHashCode(); }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var that = obj as Cid;
+            return that == null
+                ? false
+                : Encode() == that.Encode();
+        }
+
+        /// <inheritdoc />
+        public bool Equals(Cid that) { return Encode() == that.Encode(); }
+
+        /// <summary>
+        ///   Value equality.
+        /// </summary>
+        public static bool operator ==(Cid a, Cid b)
+        {
+            if (ReferenceEquals(a, b)) return true;
+            if (ReferenceEquals(a, null)) return false;
+            if (ReferenceEquals(b, null)) return false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)